hybrid/src/main/java/jetbrains/jetpad/hybrid/TokensEditPostProcessorTrait.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.hybrid;
+
+import jetbrains.jetpad.cell.Cell;
+import jetbrains.jetpad.cell.trait.CellTrait;
+import jetbrains.jetpad.event.Key;
+import jetbrains.jetpad.event.KeyEvent;
+
+class TokensEditPostProcessorTrait<SourceT> extends CellTrait {
+  private BaseHybridSynchronizer<SourceT, ?> mySync;
+  private TokensEditPostProcessor<SourceT> myPostProcessor;
+
+  TokensEditPostProcessorTrait(BaseHybridSynchronizer<SourceT, ?> sync, TokensEditPostProcessor<SourceT> postProcessor) {
+    mySync = sync;
+    myPostProcessor = postProcessor;
+  }
+
+  @Override
+  public void onKeyTyped(Cell cell, KeyEvent event) {
+    myPostProcessor.afterTokensEdit(mySync.tokens(), mySync.property().get());
+  }
+
+  @Override
+  public void onKeyPressed(Cell cell, KeyEvent event) {
+    if (isRemove(event)) {
```

Comment:
That's to exclude cases when tokens are not going to change, for example, when you hit Shift and alike, or when you type (changes will be visible in `afterTokensEdit`). I can add test which asserts tokens do change on every call.